File: backend/modules/request-group/model/request-group.go
package model

import (
	"time"
)

type RequestGroup struct {
	Id        string    `json:"id,omitempty" bson:"_id,omitempty"`
	Requests  []Request `json:"requests" bson:"requests"`
	CreateAt  time.Time `json:"createdAt" bson:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt" bson:"updatedAt"`
}

File: request-debug-front/src/app/core/models/request-group.ts
export interface RequestFile {
  filename: string;
  size: number;
}

export interface Request {
  id: string;
  method: string;
  host: string;
  url: string;
  bodySize: number;
  bodyRaw: string;
  date: string;
  ip: string;
  form: Record<string, string[]>;
  files: Record<string, RequestFile[]>;
  queryParams: Record<string, string>;
  headers: Record<string, string>;
}

export interface RequestGroup {
  id: string;
  requests: Request[];
  createdAt: string;
  updatedAt: string;
}
